addon/mixins/pagination-support-serializer.js

```diff
@@ -44,12 +45,46 @@ function extractPageNumberFromLink (link) {
     return NO_PAGE_NUMBER
   }
 
-  let url = new URL(link)
-  let page = url.searchParams.get('page')
+  let query = extractQueryFromLink(link)
+
+  if (!query) {
+    return NO_PAGE_NUMBER
+  }
+
+  let params = splitQueryIntoSeparateParams(query)
+  let paramsLength = params.length
+
+  if (paramsLength === 0) {
+    return NO_PAGE_NUMBER
+  }
+
+  let page
+
+  for (let i = 0; i < paramsLength; i++) {
```

Comment:
Nitpicky, I really don't like for loops, do you like any of the following? 😄 

```javascript
let paramString = params.find((paramString) => {
  let [key, value] = splitParamIntoKeyValue(paramString)
  return value === PAGE_PARAM_KEY
})

if (!paramString) ...

let [ key, value: page ] = splitParamIntoKeyValue(paramString)

return Number(page)
```

or more direct replacement with Array.prototype.some

```javascript
let page = null

params.some((paramString) => {
  let [key, value] = splitParamIntoKeyValue(paramString)
  if (value === PAGE_PARAM_KEY) {
    page = value
    return true
  }
  return false
})

if (!page) ...

return Number(page)
```

Reply:
🎉